crates/uv/src/target.rs

```diff
@@ -0,0 +1,142 @@
+use pep508_rs::MarkerEnvironment;
+use platform_tags::{Arch, Os, Platform};
+
+/// The supported target triples. Each triple consists of an architecture, vendor, and operating
+/// system.
+///
+/// See: <https://doc.rust-lang.org/nightly/rustc/platform-support.html>
+#[derive(Debug, Clone, Copy, Eq, PartialEq, clap::ValueEnum)]
+pub(crate) enum TargetTriple {
+    #[value(name = "x86_64-pc-windows-msvc")]
+    X8664PcWindowsMsvc,
+    #[value(name = "x86_64-unknown-linux-gnu")]
+    X8664UnknownLinuxGnu,
+    #[value(name = "x86_64-apple-darwin")]
+    X8664AppleDarwin,
+    #[value(name = "aarch64-apple-darwin")]
+    Aarch64AppleDarwin,
+    #[value(name = "aarch64-unknown-linux-gnu")]
+    Aarch64UnknownLinuxGnu,
+    #[value(name = "aarch64-unknown-linux-musl")]
+    Aarch64UnknownLinuxMusl,
+    #[value(name = "x86_64-unknown-linux-musl")]
+    X8664UnknownLinuxMusl,
```

Comment:
I don't know that this is actually the representation we want to use, but it... kind of works well?